ui/src/app/components/workflows/workflow/workflow.component.ts

```diff
@@ -107,9 +113,18 @@ export class WorkflowComponent implements OnInit, OnDestroy {
       });
   }
 
+  runWorkflow(id: number) {
+    this.runWorkflowDialogSubscription = this.confirmationDialogService
+      .confirm(ConfirmationDialogTypes.YesOrNo, texts.RUN_WORKFLOW_CONFIRMATION_TITLE, texts.RUN_WORKFLOW_CONFIRMATION_CONTENT)
+      .subscribe((confirmed) => {
+        if (confirmed) this.store.dispatch(new RunWorkflow(id));
+      });
+  }
+
   ngOnDestroy(): void {
     !!this.workflowSubscription && this.workflowSubscription.unsubscribe();
     !!this.paramsSubscription && this.paramsSubscription.unsubscribe();
     !!this.confirmationDialogServiceSubscription && this.confirmationDialogServiceSubscription.unsubscribe();
+    !!this.runWorkflowDialogSubscription ? this.runWorkflowDialogSubscription.unsubscribe() : '';
```

Comment:
I think this version looks better !!this.runWorkflowDialogSubscription && !!this.runWorkflowDialogSubscription. unsubscribe();